### DISCUSSION TITLE
[Suggestion] Namespace shorthand and naming

### DISCUSSION THREAD

**Discussion by Mayriad** [Difegue/LANraragi#279] [Ideas]
*This issue comes from #274 my big list of LRR bugs and suggestion. Please refer to that to see details of my setup and usage.*

Currently LANraragi does not support namespace shorthand in search terms, which means terms like `l:english` will not work and you have to type `language:english`. Allowing shorthand like `l:` can make it much easier to type these search terms. I understand the difficulty here since LANraragi allows custom namespaces to be created freely and also handles metadata from different websites that may use different namespaces, but this problem can be solved by abbreviating popular namespaces only e.g., `p:` will only mean `parody:` and not a custom namespace like `posted:`.

The EH metadata plugin I wrote for my script retrieves additional timestamp and uploader metadata and I have been assuming that eventually LANraragi can use the abbreviations below:
```
a - artist
c - character
f - female
g - group
m - male
l - language
r - reclass
p - parody

u - uploader
t - timestamp ("posted" in JSON returned by EH API)
i - imported (replacing "date_added" from the DateAdded metadata plugin)
d - division (replacing the default "category" namespace)
```
This way probably all namespaces can be abbreviated, but of course this is just my wishful thinking.

**Comment by Difegue** [comment]
I never thought about this before but since I'm going through old issues right now: What if a user actually creates custom namespaces that match those shorthands, like `a:thing, p:other_thing`? 

As it is it wouldn't be too hard to add specific checks for those shorthands in the Search engine and replace them, but I feel the solution would be kinda limited and not very useful?

Moving to a discussion. (I'm loving this feature already)
